packages/flutter/lib/src/widgets/two_dimensional_viewport.dart

```diff
@@ -780,7 +780,9 @@ abstract class RenderTwoDimensionalViewport extends RenderBox implements RenderA
   /// [TwoDimensionalViewportParentData], or a subclass thereof.
   @protected
   TwoDimensionalViewportParentData parentDataOf(RenderBox child) {
-    assert(_children.containsValue(child));
+    assert(_children.containsValue(child) ||
+        _keepAliveBucket.containsValue(child) ||
+        _debugOrphans!.contains(child));
```

Comment:
Yeah, I think this is definitely right, since during layout we may need to access parent data on any of the children currently being moved around. 👍